pymeasure/instruments/agilent/agilent8722ES.py

```diff
@@ -178,10 +205,42 @@ def frequencies(self):
             num=self.scan_points
         )
 
+    @property
+    def data_complex(self):
+        """ Returns the complex power from the last scan
+        """
+        # TODO: Implement binary transfer instead of ASCII
+        data = np.loadtxt(
+            BytesIO(self.ask("FORM4;OUTPDATA").encode()),
+            delimiter=',',
+            dtype=np.float32
+        )
+        data_complex = data[:, 0] + 1j * data[:, 1]
+        return data_complex
+
+    @property
+    def data_log_magnitude(self):
+        """ Returns the absolute magnitude values in dB from the last scan
+        """
+        return 20*np.log10(self.data_magnitude)
+
+    @property
+    def data_magnitude(self):
+        """ Returns the absolute magnitude values from the last scan
+        """
+        return np.abs(self.data_complex)
+
+    @property
+    def data_phase(self):
+        """ Returns the phase in degrees from the last scan
+        """
+        return np.degrees(np.angle(self.data_complex))
+    
     @property
     def data(self):
         """ Returns the real and imaginary data from the last scan
         """
+        warnings.warn("Don't use this function, use data_complex instead", FutureWarning)
```

Comment:
Thank you! Now, to round this out, change the deprecated properties/methods to use the *new* ones (if appropriate, i.e. the same result can be computed). 
This way, you avoid duplicating code (which later maybe only gets fixed in one place, not the other.
For example, here you could just say
``` 
res = self.data_complex
return res.real, res.complex
```
if I'm not deceived.